utils/sample_qc.py

```diff
@@ -0,0 +1,487 @@
+    When `binzarization_threshold ` is set, the callrate is transformed to a 0/1 value based on the thredhold.
+    E.g. with the default threshold of 0.25, all entries with a callrate < 0.25 are considered as 0s, others as 1s.
+
+    :param MatrixTable callrate_mt: Input callrate MT
+    :param float binzarization_threshold: binzarization_threshold. None is no threshold desired
+    :return: eigenvalues, scores_ht, loadings_ht
+    :rtype: (list of float, Table Table)
+    """
+    logger.info("Running platform PCA")
+
+    if binzarization_threshold is not None:
+        callrate_mt = callrate_mt.annotate_entries(callrate=hl.int(callrate_mt.callrate > binzarization_threshold))
+    # Center until Hail's PCA does it for you
+    callrate_mt = callrate_mt.annotate_rows(mean_callrate=hl.agg.mean(callrate_mt.callrate))
+    callrate_mt = callrate_mt.annotate_entries(callrate=callrate_mt.callrate - callrate_mt.mean_callrate)
+    eigenvalues, scores, loadings = hl.pca(callrate_mt.callrate, compute_loadings=True)  # TODO:  Evaluate whether computing loadings is a good / worthy thing
+    logger.info('Platform PCA eigenvalues: {}'.format(eigenvalues))
+
+    return eigenvalues, scores, loadings
+
+
+def assign_platform_from_pcs(
+        platform_pca_scores_ht: hl.Table,
+        pc_scores_ann: str = 'scores',
+        hdbscan_min_cluster_size: int = 500,
+        hdbscan_min_samples: int = None
+) -> hl.Table:
+    """
+    Assigns platforms using HBDSCAN on the results of call rate PCA.
+
+    :param Table platform_pca_scores_ht: Input table with the PCA score for each sample
+    :param str pc_scores_ann: Field containing the scores
+    :param int hdbscan_min_cluster_size: HDBSCAN `min_cluster_size` parameter
+    :param int hdbscan_min_samples: HDBSCAN `min_samples` parameter
+    :return: A Table with a `qc_platform` annotation containing the platform based on HDBSCAN clustering
+    """
+    logger.info("Assigning platforms based on platform PCA clustering")
+
+    # Read and format data for clustering
+    data = platform_pca_scores_ht.to_pandas()
+    callrate_data = np.matrix(data[pc_scores_ann].tolist())
+    logger.info('Assigning platforms to {} samples.'.format(len(callrate_data)))
+
+    # Cluster data
+    clusterer = hdbscan.HDBSCAN(min_cluster_size=hdbscan_min_cluster_size, min_samples=hdbscan_min_samples)
+    cluster_labels = clusterer.fit_predict(callrate_data)
+    n_clusters = len(set(cluster_labels)) - (-1 in cluster_labels)  # NOTE: -1 is the label for noisy (un-classifiable) data points
+    logger.info('Found {} unique platforms during platform imputation.'.format(n_clusters))
+
+    data['qc_platform'] = cluster_labels
+    ht = hl.Table.from_pandas(data, key=[*platform_pca_scores_ht.key])
+    ht = ht.annotate(qc_platform='platform_'+hl.str(ht.qc_platform))
+    return ht
+
+# TODO: This should be reviewed / merged with work from Kristen
+def impute_sex(
```

Comment:
might be good to rename the function so it's not too similar to "hl.impute_sex"

Reply:
Sure  -- any thoughts? :)

Reply:
maybe infer_sex, assign_sex, annotate_sex, or call_sex? just something so that it doesn't get confused with hail's impute_sex

Reply:
all right, I'll go with `infer_sex` since we also have `infer_families`